### PR TITLE
glog: disable broken test under clang

### PR DESCRIPTION
**Pull request by tobim**
## Description of changes

https://github.com/google/glog/issues/937

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/glog/default.nix [hunk 1]
```diff
@@ -21,22 +21,24 @@ stdenv.mkDerivation rec {
     "-DBUILD_SHARED_LIBS=ON"
   ];
 
-  # TODO: Re-enable Darwin tests once we're on a release that has https://github.com/google/glog/issues/709#issuecomment-960381653 fixed
-  doCheck = !stdenv.isDarwin;
```
Comment (r-burns) [inline pkgs/development/libraries/glog/default.nix:25]: Needs `doCheck = true;`. Removing this line has disabled tests on all platforms

**Review by r-burns — CHANGES_REQUESTED** [review]
Unfortunately I think even more failing tests have come up since I last looked at this. With `doCheck = true` on x86_64-darwin, I see failures in `logging_custom_prefix`, `logging`, and `mock-log`.

File: pkgs/development/libraries/glog/default.nix (hunk 1 above)
Comment (tobim) [reply pkgs/development/libraries/glog/default.nix:25]: I don't know why but for some reason I thought this was on by default now.

**Comment by tobim** [comment]
The latest version of the patch disables the ones you mentioned explicitly now, can you check again?

**Review by r-burns — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/glog/default.nix [hunk 2]
```diff
@@ -21,22 +21,30 @@ stdenv.mkDerivation rec {
...
       ] ++ lib.optionals stdenv.hostPlatform.isStatic [
         "LogBacktraceAt.DoesBacktraceAtRightLineWhenEnabled"
+      ] ++ lib.optionals stdenv.cc.isClang [
+        # Clang optimizes an expected allocation away.
+        # See https://github.com/google/glog/issues/937
+        "DeathNoAllocNewHook.logging"
+      ] ++ lib.optionals stdenv.isDarwin [
+        "logging_custom_prefix"
+        "logging"
+        "mock-log"
```
[the remark this answers is not in the document]
Comment (tobim) [reply pkgs/development/libraries/glog/default.nix]: Thanks, I implemented the exclusion according to your recommendation.

**Comment by tobim** [comment]
I added a workaround to a broken glogConfig.cmake installation with the latest commit.

**Review by r-burns — APPROVED** [review]
Looks good! Tested and working on x86_64-darwin. Sorry about all that ctest nonsense.

**Comment by vcunat** [comment]
Some of the tests crash on `*-darwin` now:
https://hydra.nixos.org/build/232481702/nixlog/2/tail

**Comment by tobim** [comment]
Too bad. I guess we'll have to disable again.

**Comment by tjni** [comment]
I don't know if this is the best way to disable the tests, but it suffices to skip them locally on my aarch64-darwin machine: https://github.com/NixOS/nixpkgs/pull/251729. I'm happy to iterate on this to make this better moving forward!

**Pull request merged**